src/rendering/renderengine.cpp

```diff
@@ -1061,10 +1032,31 @@ void RenderEngine::setResolveData(ghoul::Dictionary resolveData) {
 }
 
 /**
-  * Mark that one screenshot should be taken
-*/
-void RenderEngine::takeScreenShot() {
-    _shouldTakeScreenshot = true;
+ * Take a screenshot and store it in the ${SCREENSHOTS} directory
+ */
+void RenderEngine::takeScreenshot() {
+    // We only create the directory here, as we don't want to spam the users
+    // screenshot folder everytime we start OpenSpace even when we are not taking any
+    // screenshots. So the first time we actually take one, we create the folder:
+
+    if (!FileSys.directoryExists(absPath("${SCREENSHOTS}"))) {
+        FileSys.createDirectory(
+            absPath("${SCREENSHOTS}"),
+            ghoul::filesystem::FileSystem::Recursive::Yes
+        );
+    }
+
+    _latestScreenShotNumber = global::windowDelegate.takeScreenshot(_applyWarping);
```

Comment:
This is also very brittle as, for example, in the case of stereo or two windows, two screenshots will be taken, but the function will only return the number for the first screenshot.  Same as below, we need to hook into SGCT for this rather than trying to reverse engineer how many screenshots were taken

Reply:
When taking a screenshot with multiple windows, sgct uses the same screenshot number for all of them. I tested it now, and found a bug in master where two windows would try to write to the same file when taking a screenshot. Fixed it in this branch by adding the window number to the filename iff more than one window is used.

I suggest letting the screenshot function return the screenshot number instead and letting client code deal with reconstructing the file name if necessary. The alternative would be to return a list of filenames, which wont make client code happier anyways. Thoughts?